### DISCUSSION TITLE
Todays totals wrong

### DISCUSSION THREAD

**Discussion by MikealSouth** [davidrapan/ha-solarman#224] [Q&A]
### Description

For a couple of weeks/months the Integrations displays incorrect sums for "Today Production/Usage/etc"  for a device that's called "deye_12k".
Here is the "Today's Production" as a example.
![Screenshot from 2024-10-17 11-23-38](https://github.com/user-attachments/assets/46c2e032-f7fb-4756-8807-a24ce30ac1ab)

![WhatsApp Image 2024-10-17 at 11 14 58](https://github.com/user-attachments/assets/74d41bab-4af3-41a0-b2bc-d66616843f5f)

Could someone please point me to what I am doing wrong?
The WiFi connectivity is excellent, also I am using the SG04LP3 yaml file.

### Reproduction Steps

Just gathering data, the current sensor data like production, consumption, battery SOC, seems to be fine.

### Log

[home-assistant_solarman_2024-10-17T09-18-25.668Z.log](https://github.com/user-attachments/files/17409493/home-assistant_solarman_2024-10-17T09-18-25.668Z.log)

### Version

24.10.16

**Comment by davidrapan** [comment]
Hey @MikealSouth, I wouldn't think much of it cause today sensors are not important at all. It could have been just a fluke. The most important is if the total sensors worked and had values as expected.

**Comment by MikealSouth** [comment]
> Hey @MikealSouth, I wouldn't think much of it cause those sensors are not important at all. It could have been just a fluke. The most important is if the total sensors worked and had values as expected.

Hey @davidrapan, the problem is that the total sensors have the same problem, here " Total Production" as my example.
![Screenshot from 2024-10-17 11-35-37](https://github.com/user-attachments/assets/434bee4d-64c2-4afa-b0fe-beb47d941d85)

The smartphone app seems to have the correct values
![WhatsApp Image 2024-10-17 at 11 38 57](https://github.com/user-attachments/assets/45e05381-3060-411a-bbff-e50b0348739d)

**Comment by davidrapan** [comment]
So that means that your device caused it.

You will have to correct your history using Developer tools >> STATISTICS >> Total Production >> Graph icon and find the time when this occurred and fix it.

**Comment by MikealSouth** [comment]
OK, interesting, thank you.
Problem is that I can't seem to find the time that the error occured, pretty sure that this happened last month, then I look into the history of last month I can't find anything
![Screenshot from 2024-10-17 12-04-56](https://github.com/user-attachments/assets/e731c86c-8ca2-467d-9603-ffabd1ad7d2b)

I am currently trying to use the "Fix Issue" function as I seem to have tons of them.
![Screenshot from 2024-10-17 12-12-11](https://github.com/user-attachments/assets/a54a81f1-9631-407f-a99d-2f9f4f1e6544)

**Comment by davidrapan** [comment]
You have to correct it using that icon on the right. There you will see datetime picker and timestamps with values...

**Comment by MikealSouth** [comment]
OK, thank you for all of your help but I need to bug you once more but this. 
Do I just need to find one occurance of this or the first one around a month ago?
I've tried to google this but can't really find the answer, sorry to bug you teaching me how troubleshoot HA.

Here is my example, yesterdays export to grid
![Screenshot from 2024-10-17 12-44-01](https://github.com/user-attachments/assets/a4a31c21-1a27-48e3-9e49-d34c96743b1b)

I couldn't even fully charge my 14,6kWh battery, no way my system has exported +70kWh.
Do I just correct/guess these values or do I need to find the first occurance os these spikes?

Is there a way to just delete the historic values and just start from scratch?

**Comment by davidrapan** [comment]
You need to fix all occasions when this fluke happened. So it depends of how many times it happened. If you click on that bottom left button called OUTLIERS or something like that and if lucky it will show you all the occasions when irregular delta was detected and then if you fix those it should be enough.

So let's say you see for example in some time +13 kWh you correct it to 0 kWh and so on.

> Is there a way to just delete the historic values and just start from scratch?

Yes there is a way. If you delete the device from the integration and then again go into STATISTICS page and click on FIX ISSUE and then CONFIRM on all the total sensors which belong to that deleted device. You will essentially delete whole long term statistics (data used in the energy dashboard) of that particular sensor.

**Comment by MikealSouth** [comment]
I used to have several of these events a day so I chose the "delete historic data" path.
Somehow after deleting the device, dev > statistic > search for device and "fix issues" the data came back as seen as I added that device again.
So I want the more hardcore route (after a full backup) and used the SQLite explorer
```
SELECT * FROM "statistics_meta" WHERE "statistic_id" = "sensor.deye_12k_today_production";
765 sensor.deye_12k_today_production;
DELETE FROM statistics WHERE metadata_id="765";
SELECT * FROM "statistics_meta" WHERE "statistic_id" = "sensor.deye_12k_total_production";
766 sensor.deye_12k_total_production;
DELETE FROM statistics WHERE metadata_id="766";
```

I will report back if that helped.
Thank you very much @davidrapan

**Comment by davidrapan** [comment]
> Somehow after deleting the device, dev > statistic > search for device and "fix issues" the data came back as seen as I added that device again.

When you click on the sensor you see short term statistics. 😉 For energy dashboard are important long term statistics and those are deleted using this procedure.

**Comment by MikealSouth** [comment]
I totally understand your point and if I would have seen another way I would have taken that.
The only real long term statistic I really need is this:
<img width="292" alt="Screenshot 2024-10-17 at 20 31 59" src="https://github.com/user-attachments/assets/9178f3d9-c23a-4789-a6d5-590bc10ac222">

Since April 2nd my home has used 114kWh from the grid and send 4498 kWh to the grid.

Would it be great to have that data? Yes sure.
If my approach doesn't work tomorrow I'll revert back and put more time into understanding the solution.
